common/shared.c

```diff
@@ -557,11 +558,19 @@ int ft_getinfo(struct fi_info *hints, struct fi_info **info)
 	if (!hints->ep_attr->type)
 		hints->ep_attr->type = FI_EP_RDM;
 
+	prefix = hints->mode & FI_MSG_PREFIX;
```

Comment:
This doesn't allow the app to indicate that it can support MSG_PREFIX, without forcing the provider to also support it.  I think we need some sort of 'force_prefix' variable that only gets set if -k is specified.